packages/lesswrong/server/resolvers/commentResolvers.ts

```diff
@@ -112,10 +112,14 @@ defineQuery({
       return result
     }
 
-    const getUserVotesOnPoll = async (userId: string, pollCommentId: string, recommendationReason:string) => {
+    const getUserVotesOnPoll = async (userId: string, pollCommentId: string, recommendationReason:string, whoseVote:"you"|"they") => {
       const comments = await context.repos.comments.getPopularPollCommentsWithUserVotes(userId, limit * 3, pollCommentId)
       const sampled = sampleSize(comments, Math.round(limit / 2))
-      const result = sampled.map(comment => ({comment, recommendationReason: recommendationReason, yourVote: comment.yourVote, theirVote: comment.theirVote}));
+      const result = sampled.map(comment => {
+        return whoseVote === "you" 
+          ? {comment, recommendationReason: recommendationReason, yourVote: comment.userVote }
+          : {comment, recommendationReason: recommendationReason, theirVote: comment.userVote};
+      });
```

Comment:
Something about the way this is structured doesn't feel right, but it's fine to ship a fix for now.